### PR TITLE
Compute shader (cleaned)

### PR DESCRIPTION
**Pull request by JSandusky**
Cleaned cherry pick of #192

### CODE REVIEW

**Comment by eugeneko** [comment]
Damn, I forgot about this PR with all this stuff in the world...

@JSandusky Is there any reason why `URHO3D_COMPUTE` is off by default?
Would be nice to have the code covered by CI

File: Source/Urho3D/Graphics/CubemapCapture.cpp [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+#include "../Graphics/View.h"
+#include "../Graphics/Zone.h"
+
+#ifdef URHO3D_COMPUTE
+    #include "../Graphics/ComputeDevice.h"
+#endif
+
+#include "../DebugNew.h"
+
+#pragma optimize("", off)
```
Comment (eugeneko) [inline Source/Urho3D/Graphics/CubemapCapture.cpp:45]: Why needed?

Comment (JSandusky) [reply Source/Urho3D/Graphics/CubemapCapture.cpp:45]: The optimize-off is just a straggler that didn't get cleaned out after debugging, it has no legitimate reason to be there.

**Comment by JSandusky** [comment]
Not interested in reviewing further, take it or leave it (posted code MIT as described in headers).

**Comment by rokups** [comment]
Yeah we are a bit holding on to it. Eugene is doing a major engine rewrite. 
Seems to us it would be best to finalize this PR when new renderer is 
complete. Sorry for letting it hang for so long..

**Comment by eugeneko** [comment]
This PR is totally fine. I just want to do some other stuff before I merge
it.
I’ll hopefully return to it when I make new renderer for rbfx, at least in
some basic shape. You can see I have some work ongoing on top of this PR
branch.

Sorry for delay and thank you for this PR (and for GS/TS as well)

чт, 24 сент. 2020 г. в 07:15, Rokas Kupstys <notifications@github.com>:

**Comment by JSandusky** [comment]
No problem, there's nothing magical going on in there. Just the shortest route I saw for how things were at the time. Do what you have to.

**Pull request closed without merging**